src/xtopcom/xconfig/xchain_names.h

```diff
@@ -7,12 +7,17 @@
 #include "xvledger/xvaccount.h"
 
 #include <string>
+#include <map>
 
 NS_BEG2(top, config)
 
 XINLINE_CONSTEXPR char const * chain_name_mainnet{"new_horizons"};
 XINLINE_CONSTEXPR char const * chain_name_testnet{"galileo"};
 
+static std::map<std::string, uint32_t> chain_name_id_dict;
```

Comment:
not define static global variable in .h file.

better to use xnetwork_id_t (or give it an alias name, xchain_id_t).

meanwhile, any reason for using `uint32_t`?